covid19-dashboard/dc-covid19/src/App.tsx

```diff
@@ -14,186 +14,122 @@
+ * @param date
+ * @param deltaDays
+ */
+const getRealISODatesFromArrayOfDeltaDays = (
+  date: string,
+  deltaDays: number[]
+): string[] => {
+  if (date === undefined || date === null) return [];
+  const dates = deltaDays.map(deltaDate =>
+    addOrSubtractNDaysToDate(date, -deltaDate)
+  );
+  return dates.sort((a, b) => (a < b ? -1 : a > b ? 1 : 0));
+};
 
-    /**
-     * Request the data from the server, and store the Promises in that order.
-     * The order is important [cases, deaths, population, places].
-     */
-    fetchData = () => {
-        const url: string = `/api/`
-        const apis = [
-            url + 'total-cases',
-            url + 'total-deaths',
-            url + 'population',
-            url + 'places'];
-        // Map every url to the promise of the fetch
-        return apis.map(api => fetch(api));
-    }
+/**
+ * Converts a number to a string including commas for readability.
+ * For example, int(10000) would get converted to str(10,000)
+ * @param num: the number to replace
+ */
+const numberWithCommas = (num: number): string => {
+  if (!num) return '0';
+  return num.toString().replace(/\B(?=(\d{3})+(?!\d))/g, ',');
+};
 
-    /**
-     * Returns only the data corresponding to a specific region. County, State or within a State.
-     * @param allData: all the data in the form of [cases, deaths, population, names]
-     * @param region: can be State, County or a State's geoId: geoId/XX
-     */
-    getStatsDataOnlyForRegion = (allData: {}[], region: string): {}[]=> {
-        let output: {}[] = []
-        const geoIdsContainedInSelectedRegion = filterGeoIdThatBelongTo(allData[3], region)
+/**
+ * Converts an ISO date to English date.
+ * For example, 2020-01-01 is converted to January 1st, 2020.
+ * @param ISOdate: a string representing an ISO date.
+ */
+const prettifyDate = (ISOdate: string): string => {
+  const prettifiedDate = moment(ISOdate).format('MMMM Do, YYYY');
+  // If prettifiedDate is not valid, it means it wasn't a ISOdate input.
+  if (!prettifiedDate) return 'Some Day';
+  return prettifiedDate;
+};
 
-        // Iterate through all-cases and all-deaths which are found in index 0 and 1
-        for (let dataSet of this.state.allData.slice(0, 2)){
-            let filteredDataSet = {}
-            for (let date in dataSet){
-                const geoIdsForIterativeDate = dataSet[date]
-                filteredDataSet[date] = filterJSONByArrayOfKeys(geoIdsForIterativeDate, geoIdsContainedInSelectedRegion)
-            }
-            output.push(filteredDataSet)
-        }
+/**
+ * Adds or subtracts any number of days to an ISO date.
+ * For example, 2020-01-07 minus 2 days would return 2020-01-05.
+ * NOTE: You can pass in negative or positive days accordingly.
+ * @param ISOdate: a string representing a date in ISO format
+ * @param days: either positive or negative days to add
+ */
+const addOrSubtractNDaysToDate = (ISOdate: string, days: number): string => {
```

Comment:
seems you need a class to hold the date string, and have methods like:

addDays(n) // n could be positive or negative
string() // get the pretty string
makeRange(n) get range of days

Reply:
I removed prettifyString(), since it's no longer used and I added the method getDatesFromDeltaDays() to the IsoDate class.